packages/components/button/src/button.options.ts

```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@angular/core';
+
+export type ButtonVariant = 'blue' | 'grey';
+
+@Injectable({ providedIn: 'root' })
+export class ButtonOptions {
```

Comment:
Could we use `useValue` here to provide defaults?

```suggestion
export interface ButtonOptions {
  defaultVariant: ButtonVariant;
  foo?: string;
}

export cost BUTTON_DEFAULTS: ButtonOptions = {
  defaultVariant: 'blue'
};

/** @internal */
export function buttonOptionsFactory() {
  const o = new ButtonOptionsDi();
  Object.assign(o, BUTTON_DEFAULTS);
  
  return o;
}

/** @internal DI token for ButtonOptions interface */
@Injectable({
  providedIn: 'root',
  useFactory: buttonOptionsFactory
})
export class ButtonOptionsDi implements ButtonOptions {
  defaultVariant;
}
```

See https://offering.solutions/blog/articles/2018/08/17/using-useclass-usefactory-usevalue-useexisting-with-treeshakable-providers-in-angular/